ocaml/xapi/storage_migrate.ml

```diff
@@ -323,12 +323,12 @@ module Local = StorageAPI (Idl.Exn.GenClient (struct
 end))
 
 let tapdisk_of_attach_info (backend : Storage_interface.backend) =
-  let xendisks, _, _, _ =
+  let _, blockdevices, _, nbds =
     Storage_interface.implementations_of_backend backend
   in
-  match xendisks with
-  | xendisk :: _ -> (
-      let path = xendisk.Storage_interface.params in
+  match (blockdevices, nbds) with
+  | blockdevice :: _, _ -> (
+      let path = blockdevice.Storage_interface.path in
```

Comment:
This change is surprising, did `implementation_of_backend` return duplicated information previously?

Reply:
Yes, see here: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/storage_smapiv1.ml#L548-L554. The `params`, from the return value of `VDI.attach`, ends up in both places.